### PR TITLE
[docs update] fix tabs on 'installation' page

### PR DESCRIPTION
**Pull request by flarnie**
**problem:**
The old "installation" page relied on HTML with inline CSS and JS, and the JS doesn't work with our new site infrastructure.

Specifically, there were some tabs which switched the content and they are not working in the new docs site.

We are trying to make minimal changes to the markdown docs files, because we want to minimize merge conflicts and also not interfere with an ongoing i18n project.

**solution:**
Made minimal changes to the installation docs file in 'docs';
- removed the non-functional inline JS
- changed a couple of 'h2' to 'h3' (this made the styles look correct)
- changed 'block' tags to 'section' and added wrapper divs (this made it render correctly)

Then we made a hacky custom component and moved the tab JS there, mostly unchanged.

Ideally we do a follow-up pass and completely move the styles and JS out of 'docs/docs/installation.md' and clean this up. But for now it's similar to other work-arounds in place for this docs rewrite.

### CODE REVIEW

**Comment by flarnie** [comment]
screen shot 2017-09-01 at 4 37 34 pm

File: www/src/templates/docs.js [hunk 1]
```diff
@@ -30,13 +31,25 @@ const sectionList = sectionListA
...
 
-const Docs = ({data, location}) => (
-  <MarkdownPage
-    location={location}
-    markdownRemark={data.markdownRemark}
-    sectionList={sectionList}
-  />
-);
+const Docs = ({data, location}) => {
+  console.log('location is ', location);
```
Comment (bvaughn) [inline www/src/templates/docs.js]: Whoops ^

File: www/src/templates/docs.js [hunk 2]
```diff
@@ -30,13 +31,25 @@ const sectionList = sectionListA
...
+  if (location.pathname === '/docs/installation.html') {
+    console.log('about to render InstallationPage');
```
Comment (bvaughn) [inline www/src/templates/docs.js]: Whoops ^

File: www/src/templates/docs.js [hunk 3]
```diff
@@ -30,13 +31,25 @@ const sectionList = sectionListA
...
-const Docs = ({data, location}) => (
-  <MarkdownPage
-    location={location}
-    markdownRemark={data.markdownRemark}
-    sectionList={sectionList}
-  />
-);
+const Docs = ({data, location}) => {
+  console.log('location is ', location);
+  if (location.pathname === '/docs/installation.html') {
```
Comment (bvaughn) [inline www/src/templates/docs.js]: This _works_ but I think we'd be better off doing this check in `gatsby-node` like we do [for the Home page](https://github.com/facebook/react/blob/gatsby/www/gatsby-node.js#L63-L70).

File: www/src/templates/docs.js (hunk 1 above)
Comment (flarnie) [reply www/src/templates/docs.js]: :) fixed while you were reviewing

File: www/src/templates/docs.js (hunk 3 above)
Comment (flarnie) [reply www/src/templates/docs.js]: Cool - will look into that.

File: www/src/templates/installation.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    document.getElementById(prevTab).focus();
+    display('target', prevTab);
+  }
+  // right arrow ->
+  if (e.keyCode === 39) {
+    document.getElementById(nextTab).focus();
+    display('target', nextTab);
+  }
+}
+window.keyToggle = keyToggle;
```
Comment (bvaughn) [inline www/src/templates/installation.js]: Whoops! I missed this.

We can't have `window` references in code that's auto-executed as part of an import b'c they break the Gatsby `build` step. I'll remove this one! 😄

File: www/src/templates/installation.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+              <div
+                css={{
+                  flex: '0 0 200px',
+                  marginLeft: 'calc(9% + 40px)',
+                }}>
+                <StickySidebar
+                  defaultActiveSection={findSectionForPath(
+                    location.pathname,
+                    sectionList,
+                  )}
```
Comment (bvaughn) [inline www/src/templates/installation.js]: Same here as with `window`. Should be:
```js
defaultActiveSection={
  location != null
    ? findSectionForPath(location.pathname, sectionList)
    : null
}
```

Comment (flarnie) [reply www/src/templates/installation.js]: Ok - good to know!

**Pull request merged**